File: src/main/scala/io/izzel/taboolib/module/ai/SimpleAi.java
package io.izzel.taboolib.module.ai;

/**
 * 实体 AI 实现接口
 *
 * @author sky
 * @since 2018-09-19 19:42
 */
public abstract class SimpleAi {

    /**
     * @return 是否开始执行
     */
    public abstract boolean shouldExecute();

    /**
     * @return 是否继续执行
     */
    public boolean continueExecute() {
        return shouldExecute();
    }

    /**
     * 开始时
     */
    public void startTask() {
    }

    /**
     * 重置时
     */
    public void resetTask() {
    }

    /**
     * 更新时
     */
    public void updateTask() {
    }
}

File: module/module-ai/src/main/kotlin/taboolib/module/ai/SimpleAi.kt
package taboolib.module.ai

/**
 * 实体 AI 实现接口
 *
 * @author sky
 * @since 2018-09-19 19:42
 */
abstract class SimpleAi {

    /**
     * @return 是否开始执行
     */
    abstract fun shouldExecute(): Boolean

    /**
     * @return 是否继续执行
     */
    open fun continueExecute(): Boolean {
        return shouldExecute()
    }

    /**
     * 开始时
     */
    open fun startTask() {
    }

    /**
     * 重置时
     */
    open fun resetTask() {
    }

    /**
     * 更新时
     */
    open fun updateTask() {
    }
}